### PR TITLE
cleanup files after test_execOrphanedFile

### LINKED ISSUE #18885: Improve tests stability for DownloadInstrumentTest
test_execOrphanedFile: files not cleaned up

20:16:27 491/1746 Test #719: DataHandlingTest_DownloadInstrumentTest ......................***Failed    4.55 sec
20:16:27 FrameworkManager-[Notice] Welcome to Mantid 3.9.20170215.1925
20:16:27 FrameworkManager-[Notice] Please cite: http://dx.doi.org/10.1016/j.nima.2014.07.029 and this release: http://dx.doi.org/10.5286/Software/Mantid
20:16:27 Running 4 tests.DownloadInstrument-[Notice] DownloadInstrument started
20:16:27 DownloadInstrument-[Notice] Downloading 2 files from the instrument repository
20:16:27 DownloadInstrument-[Notice] DownloadInstrument successful, Duration 1.75 seconds
20:16:27 ..DownloadInstrument-[Notice] DownloadInstrument started
20:16:27 DownloadInstrument-[Error] Error in execution of algorithm DownloadInstrument
20:16:27 DownloadInstrument-[Error] Unable to parse server JSON file "C:\Users\builder\AppData\Roaming\mantidproject\mantid\instrument\github.json"
20:16:27 
20:16:27 In DownloadInstrumentTest::test_execOrphanedFile:
20:16:27 C:/Jenkins/workspace/pull_requests-win7/Framework/DataHandling/test/DownloadInstrumentTest.h:160: Error: Assertion failed: alg.isExecuted()
20:16:27 C:/Jenkins/workspace/pull_requests-win7/Framework/DataHandling/test/DownloadInstrumentTest.h:143: Error: Test failed: The expected number of files downloaded was wrong.
20:16:27 C:/Jenkins/workspace/pull_requests-win7/Framework/DataHandling/test/DownloadInstrumentTest.h:143: Error: Expected (runDownloadInstrument() == 2), found (0 != 2)
20:16:27 Failed 1 of 4 tests
20:16:27 Success rate: 75%

### PR DESCRIPTION
**Pull request by NickDraper**
A test was not cleaning up after itself

**To test:**

1. all unit tests should pass
1. run test with `ctest -C Debug -R DownloadInstrumentTest -V` (-C only needed on windows)
1. Check if you have a subdirectory of "TEMPORARY_unitTest"  or any of the following files in the download directory
   - github.json
   - NewFile.xml
   - UpdatableFile.xml
   - Orphaned_Should_not_be_here.xml

Download directory:
   - win `%appdata%\mantidproject\mantid\instrument`
   - others `$home/.mantid/instrument`

Fixes #18885.

### RELEASE NOTES

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AndreiSavici** [comment]
Works as described, but it has a side effect that it erases any files you have in that directory. For example, with a clean instrument directory, I run `DownloadInstrument` in MantidPlot. I get an updated file for `Bilby_Definition.xml`. After I run the test, the file is gone

**Comment by NickDraper** [comment]
The test itself only deletes files it creates, if other files were removed then it was because DownloadInstrument decided that they were orphaned files and not needed anymore.

It is not a bad idea to perform the tests in another directory though.

File: Framework/Kernel/src/ConfigService.cpp [hunk 1]
```diff
@@ -1622,6 +1622,15 @@ const std::vector<std::string> &ConfigServiceImpl::getUserSearchDirs() const {
 }
 
 /**
+* Sets the search directories for XML instrument definition files (IDFs)
+* @params directories An ordered list of paths for instrument searching
```
Comment (AndreiSavici) [inline Framework/Kernel/src/ConfigService.cpp]: `@param` without `s`, otherwise Doxygen warning

**Comment by AndreiSavici** [comment]
Works as described, and it does not change my downloaded files. :shipit: when the builds pass

**Pull request merged**